src/shogun/mathematics/graph/ops/shogun/Packet.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#ifndef SHOGUN_PACKET_SHOGUN_H_
+#define SHOGUN_PACKET_SHOGUN_H_
+
+#include "details/PacketType.h"
+
+namespace shogun
+{
+	namespace graph
+	{
+		namespace op
+		{
+			// template <typename T>
+			// aligned_vector load_sse(void* input1);
+
+			template <typename T>
+			aligned_vector load_avx(void* input1);
+
+			// template <typename T>
+			// aligned_vector load_avx512f(void* input1);
+
+			// template <typename T>
+			// void store_sse(const aligned_vector& input1, void* output);
+
+			template <typename T>
+			void store_avx(const aligned_vector& input1, void* output);
+
+			// template <typename T>
+			// void store_avx512f(const aligned_vector& input1, void* output);
+
+			enum class RegisterType
+			{
+				SSE = 16,
+				AVX = 32,
+				AVX512 = 64
+			};
+
+			struct Packet
```

Comment:
@karlnapf basically this will be the abstraction of the data on the stack. And then the respective implementation (depending on CPU architecture) is called and gets the value from the variant. This actually works on my laptop...